rating_curves/build_rating_curve.R

```diff
@@ -73,63 +71,36 @@ CheckLoad <- function(path, filename) {
 # =======================================
 # Read-in stage and flow data
 
+#*****
+# TODO(awlostowski): Look for data, if it doesn't exist make it/get it
+#*****
+
 # Read in Rock Report stage observations at Pineview
 rockreport.path     <- here::here("data","rock_report")
 rockreport.filename <- "stage_poudre_rock_report.RDS"
-rock.stage          <- CheckLoad(rockreport.path, rockreport.filename)
+stage          <- CheckLoad(rockreport.path, rockreport.filename)
+if (stage == FALSE) {
+  logger::error('Please run rock_stage_webscrape.R to get stage data')
+  stop()
+}
   
 # Read in gage observations at Poudre Park
 gagedata.path     <- here::here("data","gauge")
 gagedata.filename <- "poudre_park_flow.RDS"
-gage.flow <- CheckLoad(gagedata.path, gagedata.filename)
-
-# =======================================
-# Align datetime conventions of each dataset and join
-
-# specify stage observation date and time
-stage <- rock.stage %>%
-  mutate(
-    time = case_when(
-      str_detect(title, regex('noon', ignore_case = T)) == TRUE ~ "12:00",
-      str_detect(title, regex('noon', ignore_case = T)) == FALSE ~ time
+flow <- CheckLoad(gagedata.path, gagedata.filename)
+if (flow == FALSE) {
+  logger::log_info(
+    'Attempting to get Poudre Park flow data from FoCo OpenData...'
     )
-  ) %>%
-  mutate(
-    time = format(
-      round(
-        strptime(paste("2001-01-01", time), format="%Y-%m-%d %H:%M"), 
-            units="hours"), 
-      format="%H:%M")) %>% 
-  filter(is.na(time) == F) %>% 
-  mutate(date = as.POSIXct(paste(as.character(date), time),
-                           format = "%Y-%m-%d %H:%M"))
-
-# round flow time to nearest hour, make date variable full datetime, 
-# and average flow observations within the same hour
-flow <- gage.flow %>% 
-  mutate(time = format(
-            round(
-              strptime(
-                paste("2001-01-01", gage.flow$time),
-                format="%Y-%m-%d %H:%M"
-                ), 
-              units="hours"
-              ), 
-            format="%H:%M"),
-         date = as.POSIXct(
-           paste(as.character(date), time),
-           format = "%Y-%m-%d %H:%M"
-           )
-         ) %>% 
-  group_by(date) %>% 
-  summarize(
-    flow_cfs = mean(flow_cfs, na.rm = T)
-  )
```

Comment:
all of this time munging isn't needed here anymore, because it is confined to upstream functions.